src/app/scripts/cac/control/cac-control-sidebar-directions.js

```diff
@@ -168,6 +171,24 @@ CAC.Control.SidebarDirections = (function (_, $, Control, BikeModeOptions, Geoco
             arriveBy: arriveBy
         };
 
+        // add intermediatePlaces if user edited route
+        if (waypoints) {
+            // intermediatePlaces are ordered. see comment on `getGraphPathsConsideringIntermediates`:
+            // https://github.com/opentripplanner/OpenTripPlanner/blob/master/src/main/java/org/opentripplanner/routing/impl/GraphPathFinder.java#L241
+            // if `arriveBy` is `true`, places are expected to be ordered right-to-left.
+            if (arriveBy) {
+                waypoints.reverse();
```

Comment:
I read that comment to mean that they're going to do the calculation in reverse internally but not that the waypoint arguments should be fed in in reverse.  And it looks like what the code does is build the same array either way then loop through it forward for other cases or backward for arriveBy.

Also when I add wacky waypoints to a route they appear to be traversed in the order I expect when I comment out this `waypoints.reverse()`.